server/controllers/stock/index.js

```diff
@@ -298,96 +298,91 @@ async function createInventoryAdjustment(req, res, next) {
     // pass reverse operations
     const trx = db.transaction();
 
-    const positiveAdjustmentUuid = uuid();
-    const negativeAdjustmentUuid = uuid();
+    const uniqueAdjustmentUuid = uuid();
 
-    // get all lots with positive quantities
-    const positiveQuantities = lots.filter(lot => lot.oldQuantity > 0);
+    let countNeedIncrease = 0;
+    let countNeedDecrease = 0;
+    const adjustmentLog = [];
 
-    // get all lots with negative quantities
-    // negative quantities occurs during some extra stock exit when quantity in stock
-    // is already under or equal to zero
-    const negativeQuantities = lots.filter(lot => lot.oldQuantity < 0);
+    lots.forEach(lot => {
+      const difference = lot.quantity - lot.oldQuantity;
 
-    positiveQuantities.forEach(lot => {
-      const reverseMovementObject = {
-        uuid : db.bid(uuid()),
-        lot_uuid : db.bid(lot.uuid),
-        depot_uuid : db.bid(movement.depot_uuid),
-        document_uuid : db.bid(negativeAdjustmentUuid),
-        quantity : lot.oldQuantity,
-        unit_cost : lot.unit_cost,
-        date : new Date(movement.date),
-        entity_uuid : movement.entity_uuid,
-        is_exit : 1,
-        flux_id : core.flux.INVENTORY_RESET,
-        description : movement.description,
-        user_id : req.session.user.id,
-        period_id : periodId,
-      };
-      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
-    });
-
-    negativeQuantities.forEach(lot => {
-      const reverseMovementObject = {
-        uuid : db.bid(uuid()),
-        lot_uuid : db.bid(lot.uuid),
-        depot_uuid : db.bid(movement.depot_uuid),
-        document_uuid : db.bid(positiveAdjustmentUuid),
-        quantity : lot.oldQuantity * -1,
-        unit_cost : lot.unit_cost,
-        date : new Date(movement.date),
-        entity_uuid : movement.entity_uuid,
-        is_exit : 0,
-        flux_id : core.flux.INVENTORY_RESET,
-        description : movement.description,
-        user_id : req.session.user.id,
-        period_id : periodId,
-      };
-      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
+      if (difference < 0) {
```

Comment:
It looks like there is a lot of shared/repeated information between these two `if`s.  Could you unify them a bit?

```js
const movementObject = {
          uuid : db.bid(uuid()),
          lot_uuid : db.bid(lot.uuid),
          depot_uuid : db.bid(movement.depot_uuid),
          document_uuid : db.bid(uniqueAdjustmentUuid),
          unit_cost : lot.unit_cost,
          quantity: Math.abs(quantity),
          date : new Date(movement.date),
          entity_uuid : movement.entity_uuid,
          flux_id : core.flux.INVENTORY_ADJUSTMENT,
          description : movement.description,
          user_id : req.session.user.id,
          period_id : periodId,
};

if (difference < 0) {
        countNeedDecrease++;
       Object.assign(movementObject, { is_exit : 1 });
       adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 1 }));
} else {
        countNeedIncrease++;
       Object.assign(movementObject, { is_exit : 0 });
       adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 0 }));
}

trx.addQuery('INSERT INTO stock_movement SET ?', movementObject);

```

You could actually probably get it smaller than this by adopting my proposal to remove `_.assign()`.  I think the only thing that changes is `is_exit`, right?